src/xscen/diagnostics.py

```diff
@@ -554,24 +563,29 @@ def measures_improvement(
         )
     ds1 = meas_datasets[0]
     ds2 = meas_datasets[1]
+    if dim is not None:
+        dims = [dim] if isinstance(dim, str) else dim
+        for v in ds1.data_vars:
+            if set(dims).issubset(set(ds1[v].dims)) is False:
+                raise ValueError(
+                    f"Dimension provided `dim` ({dim}) must be in every variable of `meas_datasets`"
+                )
+
     percent_better = []
     for var in ds2.data_vars:
+        if dim is None:
+            # reduce all dimensions (which may be variable dependent)
+            dims = ds2[var].dims
         if "xclim.sdba.measures.RATIO" in ds1[var].attrs["history"]:
             diff_bias = abs(ds1[var] - 1) - abs(ds2[var] - 1)
         else:
             diff_bias = abs(ds1[var]) - abs(ds2[var])
-        diff_bias = diff_bias.values.ravel()
-        diff_bias = diff_bias[~np.isnan(diff_bias)]
+        total_improved = (diff_bias >= 0).sum(dim=dims)
+        total_notnull = diff_bias.notnull().sum(dim=dims)
+        percent_better_var = total_improved / total_notnull
+        percent_better.append(percent_better_var.expand_dims({"properties": [var]}))
 
-        total = ds2[var].values.ravel()
-        total = total[~np.isnan(total)]
```

Comment:
I realized that I used `diff_bias.notnull().sum(dim=dims)`. If `diff_bias` has more null values than `ds2[var]` (i.e.  `ds1[var]` set of coordinates with null values is not a subset of those of `ds2[var]`), then this could make a difference. 

I could write `total_notnull = ds2[var].notnull().sum(dim=dims)` although I think in the end, we might prefer `diff_bias.notnull`. Thoughts?

Reply:
I think the current definition makes more sense, but it will be breaking, so I'd like to validate the change